packages/workbox-background-sync/src/lib/request-manager.js

```diff
@@ -42,49 +42,66 @@ class RequestManager {
   }
 
   /**
-   * function to start playing requests
-   * in sequence
-   * @return {void}
+   * function to play one single request given its hash
+   *
+   * @param {String} hash
+   *
+   * @return {Promise} Resolves if the request corresponding to the hash is
+   * played successfully, rejects if it fails during the replay
    *
    * @memberOf RequestManager
    * @private
    */
-  replayRequests() {
-    return this._queue.queue.reduce((promise, hash) => {
-      return promise
-        .then(async (item) => {
-          const reqData = await this._queue.getRequestFromQueue({hash});
-          if(reqData.response) {
-            // check if request is not played already
-            return;
-          }
-
-          const request = await getFetchableRequest({
-            idbRequestObject: reqData.request,
-          });
-
-          return fetch(request)
-            .then((response)=>{
-              if(!response.ok) {
-                return Promise.resolve();
-              } else {
-                // not blocking on putResponse.
-                putResponse({
-                  hash,
-                  idbObject: reqData,
-                  response: response.clone(),
-                  idbQDb: this._queue.idbQDb,
-                });
-                this._globalCallbacks.onResponse
-                  && this._globalCallbacks.onResponse(hash, response);
-              }
-            })
-            .catch((err)=>{
-              this._globalCallbacks.onRetryFailure
-                && this._globalCallbacks.onRetryFailure(hash, err);
-            });
+  async replayRequest(hash) {
+    try {
+      const reqData = await this._queue.getRequestFromQueue({hash});
+      if(reqData.response) {
+        return;
+      }
+      const request = await getFetchableRequest({
+        idbRequestObject: reqData.request,
+      });
+      const response = await fetch(request);
+      if(!response.ok) {
+        return Promise.reject(response);
+      } else {
+        // not blocking on putResponse.
+        putResponse({
+          hash,
+          idbObject: reqData,
+          response: response.clone(),
+          idbQDb: this._queue.idbQDb,
         });
-    }, Promise.resolve());
+        if (this._globalCallbacks.onResponse)
+          this._globalCallbacks.onResponse(hash, response);
+      }
+    } catch(err) {
+      return Promise.reject(err.stack);
+    }
+  }
+
+  /**
+   * function to start playing requests in sequence
+   *
+   * @return {Promise} Resolves if all requests are played successfully,
+   * rejects if any of the request fails during the replay
+   *
+   * @memberOf RequestManager
```

Comment:
This member of isn't needed for individual functions. The function is part of the class and whatever the class is a memberof is the important bit.

All of this is unnecessary because you've marked this class as private.